R/loocv.R

```diff
@@ -157,10 +158,101 @@ loocv <- function(data, return_models = FALSE, ...) {
   return(out)
 }
 
+#' Check if something is a baggr_cv object
+#' @param x object to check
+is.baggr_cv <- function(x) {
+  inherits(x, "baggr_cv")
+}
+
+#' Compare fitted models on loo
+#' @param x An object of class "baggr_cv" or a list of such objects.
+#' @param ... Additional objects of class "baggr_cv"
+#' @export loo_compare
+#' @examples
+#' # 2 models with more/less informative priors
+#' cv_1 <- loocv(schools, model = "rubin", pooling = "partial")
+#' cv_2 <- loocv(schools, model = "rubin", pooling = "partial", prior_hypermean = normal(0, 5), prior_hypersd = cauchy(0,4))
+#' loo_compare(cv_1, cv_2)
+#' @export
+loo_compare <- function(x, ...) {
+  UseMethod("loo_compare")
+}
+
+#' @aliases loo_compare
+#' @export
+loo_compare.baggr_cv <- function(x, ...) {
+  if (is.baggr_cv(x)) {
+    dots <- list(...)
+    loos <- c(list(x), dots)
+  } else {
+    if (!is.list(x) || !length(x)) {
+      stop("'x' must be a list if not a 'loo' object.")
+    }
+    loos <- x
+  }
+  if (!all(sapply(loos, is.baggr_cv))) {
+    stop("All inputs should have class 'baggr_cv'.")
+  }
+  if (length(loos) <= 1L) {
+    stop("'loo_compare' requires at least two models.")
+  }
+  Ns <- sapply(loos, function(x) nrow(x$df))
+  if (!all(Ns == Ns[1L])) {
+    stop("Not all models have the same number of data points.")
+  }
```

Comment:
Stuff like this could have rudimentary tests. You can then check with covr that you have good code coverage for `loo_compare`

Reply:
Totally agree, should be easy to put together.

Reply:
Added tests and pushed. Sorry for the delay, been crazy with the holidays.